### PR TITLE
Properly set owningPipe on CoerceTo

### PR DESCRIPTION
**Pull request by OliviaYtterbrink**
The rewriter of CoerceTo didn't rewrite the inner expression.

### CODE REVIEW

File: community/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/PatternComprehensionAcceptanceTest.scala [hunk 1]
```diff
@@ -21,9 +21,41 @@ package org.neo4j.internal.cypher.acceptance
...
+    val n2 = createLabeledNode("User")
+    val r = relate(n2, n1, "POSTED")
+
+    val query = """MATCH(t:Tweet) WITH t LIMIT 1
+                  |WITH collect(t) AS tweets
+                  |WITH [ tweet IN tweets | [ (tweet)<-[:POSTED]-(user) | user] ] AS pattern
+                  |RETURN test.toSet(pattern) AS users""".stripMargin
+
+    val result = executeWithCostPlannerOnly(query)
+    println(result.toList)
```
Comment (pontusmelke) [inline community/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/PatternComprehensionAcceptanceTest.scala]: remove the `println` and do some actual assertion on `result`

File: community/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/PatternComprehensionAcceptanceTest.scala [hunk 2]
```diff
@@ -21,9 +21,41 @@ package org.neo4j.internal.cypher.acceptance
...
+    val n1 = createLabeledNode("Tweet")
+    val n2 = createLabeledNode("User")
+    val r = relate(n2, n1, "POSTED")
+
+    val query = """MATCH(t:Tweet) WITH t LIMIT 1
+               |WITH collect(t) AS tweets
+               |RETURN test.toSet([ tweet IN tweets | [ (tweet)<-[:POSTED]-(user) | user] ]) AS users""".stripMargin
+
+    val result = executeWithCostPlannerOnly(query)
+    println(result.toList)
```
Comment (pontusmelke) [inline community/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/PatternComprehensionAcceptanceTest.scala]: remove the println and do some actual assertion on result

**Review by pontusmelke — CHANGES_REQUESTED** [review]
Nice catch! Just two minor things in the tests.

**Comment by craigtaverner** [comment]
Replaced by https://github.com/neo4j/neo4j/pull/9389

**Pull request closed without merging**